device/mellanox/x86_64-mlnx_lssn2700-r0/plugins/sfputil.py

```diff
@@ -10,6 +10,34 @@
 except ImportError as e:
     raise ImportError("%s - required module not found" % str(e))
 
+def determine_in_docker():
+    with open('/proc/self/cgroup') as f:
+        lines = [line.rstrip('\n') for line in f]
+    for line in lines:
+        words = line.split('/')
+        if len(words) > 1 and words[1] == 'docker':
+            return True
+
+    return False
+
+INSIDE_DOCKER = determine_in_docker()
+
+if INSIDE_DOCKER:
+    from swsscommon import swsscommon
```

Comment:
>swsscommon [](start = 9, length = 10)

Changing behaviors based on library availability or docker environment is not a good idea.

Reply:
revised. the swsscommon only imported in the new add function 'get_transceiver_change_event', since it will only be called from pmon container.